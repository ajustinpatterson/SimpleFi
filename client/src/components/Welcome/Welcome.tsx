import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { RouteProps } from "react-router";
import simpleFiSplash from '../../assets/images/simpleFi-splash3.svg';
import './Welcome.css';
import Footer from '../Footer/Footer';

interface WelcomeProps {
  connect: (ev: React.MouseEvent<HTMLButtonElement>) => void;
}

export default function Welcome ({ connect }: WelcomeProps) {

  const br = '\n';

  return (
    <>
    <div className='welcome'>
      <div className='welcome-splash'>
        <div className="splash-text">
          <h1>SimpleFi</h1>
          <h2>Decentralised finance investing made easy!</h2>
        </div>
        <div className="splash-connect">
          <Link to='/dashboard'>
            <button className='welcome-button' onClick={connect}>Connect wallet</button>
          </Link>
        </div>
      </div>
      <div className="welcome-media">
        <img src={simpleFiSplash} alt="Welcome to SimpleFi" className="welcome-media-image"/>
      </div>
    </div>
    <Footer/>
    </>
  )
}