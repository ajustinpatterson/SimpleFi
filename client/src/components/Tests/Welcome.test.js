import React from "react";
import {
  act,
  render,
  wait,
  fireEvent,
  screen
} from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import Welcome from "../Welcome/Welcome"
import '@testing-library/jest-dom/extend-expect'
import renderer from "react-test-renderer"

describe("Welcome", () => {

  it("renders correctly", () => {
    const tree = renderer.create(
    <BrowserRouter>
      <Welcome />
    </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
  

  it("runs the connect prop a single time when the Link element is clicked", () => {
    const onClick = jest.fn()
    render(
      <BrowserRouter>
        <Welcome connect={onClick}/>
      </BrowserRouter>
    );
    
    fireEvent.click(screen.getByText(/connect wallet/i))

    expect(onClick).toHaveBeenCalledTimes(1);
  })

})