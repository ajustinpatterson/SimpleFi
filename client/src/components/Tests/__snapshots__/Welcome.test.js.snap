// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly 1`] = `
Array [
  <div
    className="welcome"
  >
    <div
      className="welcome-splash"
    >
      <div
        className="splash-text"
      >
        <h1>
          SimpleFi
        </h1>
        <h2>
          Decentralised finance investing made easy!
        </h2>
      </div>
      <div
        className="splash-connect"
      >
        <a
          exact={true}
          href="/dashboard"
          onClick={[Function]}
        >
          <button
            className="welcome-button"
          >
            Connect wallet
          </button>
        </a>
      </div>
    </div>
    <div
      className="welcome-media"
    >
      <img
        alt="Welcome to SimpleFi"
        className="welcome-media-image"
        src="simpleFi-splash3.svg"
      />
    </div>
  </div>,
  <footer>
    <span>
      © 2020 SimpleFi
    </span>
    <span>
      ·
    </span>
    <span>
      Privacy Policy
    </span>
    <span>
      ·
    </span>
    <span>
      Twitter
    </span>
    <span>
      ·
    </span>
    <span>
      Telegram
    </span>
  </footer>,
]
`;
